src/main/webapp/scripts/map.js

```diff
@@ -80,12 +88,79 @@ window.initMap = function () {
         strokeOpacity: 0.8,
         strokeWeight: 2
       });
+      // Once cards are retrieved...
+      // Add vendor's markers
+      addVendorsToMap(map);
     },
-    (error) => {
-      if (error.code == error.PERMISSION_DENIED)
-        alert('Permission denied to access location');
-    });
+      (error) => {
+        if (error.code == error.PERMISSION_DENIED)
+          alert('Permission denied to access location');
+      });
   } else {
     alert('This browser does not support location');
   }
+};
+
+/**
+ * Function that adds vendor marks to the map
+ */
+const addVendorsToMap = (map) => {
+  const salecardTemplate = document.getElementById('salecard-template');
+  const salecardsContainer = document.getElementById('salecards-container');
+
+  //Clean previously retrieved cards
+  salecardsContainer.innerHTML = '';
+
+  //Display number of vendor found
+  displayNumberOfVendors(vendors);
+
+  Object.keys(vendors).forEach(vendorNumber => {
+    let vendor = vendors[vendorNumber];
+    let salecard = vendor.saleCard;
+
+    createModal(vendor, salecard, salecardTemplate, salecardsContainer);
+
+    var marker = new google.maps.Marker({
+      map: map,
+      position: {
+        lat: salecard.location.salepoint.lat,
+        lng: salecard.location.salepoint.lng
+      },
+      title: salecard.businessName
+    });
+
+    // Toggles salecard's modal when vendor's marker is clicked
+    marker.addListener('click', () => {
+      $(`#modal${vendor.id}`).modal('toggle');
+    })
+  });
+};
+
+/**
+ * Function that creates a modal given a salecard
+ */
+const createModal = (vendor, salecard, salecardTemplate, salecardsContainer) => {
+  let salecardCloned = salecardTemplate.content.cloneNode(true);
+
+  salecardCloned.getElementById('business-title').innerHTML = salecard.businessName;
+  salecardCloned.getElementById('business-name').innerHTML = salecard.businessName;
+  salecardCloned.getElementById('business-description').innerHTML = salecard.description;
+  salecardCloned.getElementById('vendor-name').innerHTML = `${vendor.firstName} ${vendor.lastName}`;
+  salecardCloned.getElementById('vendor-phone').innerHTML = vendor.phoneNumber;
```

Comment:
> I think we shouldn't use `innerHTML` to set the content, because of cross-site scripting attacks. Instead use: `textContent`. [This is what I found about it](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#Security_considerations)